text/0097-log-data-model.md

```diff
@@ -0,0 +1,968 @@
+value of `resource`. Can contain for example information about the application
+that emits the record or about the infrastructure where the application runs.
+Data formats that represent this data model may be designed in a manner that
+allows the `resource` field to be recorded only once per batch of log records
+that come from the same source. SHOULD follow OpenTelemetry
+[semantic conventions for Resources](https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions).
+
+### Field: `attributes`
+
+Type: key/value pair list.
+
+Description: Additional information about the specific event occurrence. "value"
+of each pair is of `any` type. Unlike the `resource` field, which is fixed for a
+particular source, `attributes` can vary for each occurrence of the event coming
+from the same source. Can contain information about the request context (other
+than TraceId/SpanId). SHOULD follow OpenTelemetry
+[semantic conventions for Attributes](https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/trace/semantic_conventions).
+
+## Example Log Records
+
+Below are examples that show one possible representation of log records in JSON.
+These are just examples to help understand the data model. Don’t treat the
+examples as _the_ way to represent this data model in JSON.
+
+This document does not define the actual encoding and format of the log record
+representation. Format definitions will be done in separate OTEPs (e.g the log
+records may be represented as msgpack, JSON, Protocol Buffer messages, etc).
+
+Example 1
+
+```json
+{
+  "timestamp": 1586960586000, // JSON needs to make a decision about 
+                              // how to represent nanoseconds.
+  "attributes": {
+    "http.status_code": 500,
+    "http.url": "http://example.com",
+    "my.custom.application.tag": "hello",
+  },
+  "resource": {
+    "service.name": "donut_shop",
+    "service.version": "semver:2.0.0",
+    "k8s.pod.uid": "1138528c-c36e-11e9-a1a7-42010a800198",
+  },
+  "trace_id": "f4dbb3edd765f620", // this is a byte sequence 
+                                  // (hex-encoded in JSON)
+  "span_id": "43222c2d51a7abe3",
+  "severity": "INFO",
+  "body": "20200415T072306-0700 INFO I like donuts"
+}
+```
+
+Example 2
+
+```json
+{
+  "timestamp": 1586960586000,
+  ...
+  "body": {
+    "i": "am",
+    "an": "event",
+    "of": {
+      "some": "complexity"
+    }
+  }
+}
+```
+
+## Open Questions
+
+- Should we store entire
+  [W3C Trace Context](https://www.w3.org/TR/trace-context/), including
+  `traceparent` and `tracestate` fields instead of only `trace_flags`?
+  
+- Is `severity_text`/`severity_number` fields design good enough?
+
+- Early draft of this proposal specified that `timestamp` should be populated
+  from a monotonic, NTP-synchronized source. I removed this requirement to avoid
+  confusion. Do we need any requirements for timestamp sources?
```

Comment:
I don't think log interchange formats should be putting requirements on the systems they run on. 

1. Is the spec going to define how conforming implementations should determine that the source meets those requirements? 
2. Is the spec going to define what the fallback process is if it doesn't meet these requirements?  

If the spec isn't going to answer 1 or 2 above, it shouldn't put the requirement in place.

Reply:
I don't think this OTEP wants to be in the business of defining those requirements.

Reply:
Agreed.

Reply:
I agree.

Reply:
Alternative approach is to use something like [RDC 5424 `timeQuality` field](https://tools.ietf.org/html/rfc5424#section-7.1).

Reply:
@hauleth good point. timeQuality can have an equivalent in OpenTelemetry semantic conventions and be recorded in the `Resource` or `Attributes`.

Reply:
Resolving as it appears everyone is in agreement.